browser/extensions/api/brave_rewards_api.cc

```diff
@@ -75,6 +75,135 @@ BraveRewardsOpenBrowserActionUIFunction::Run() {
+BraveRewardsGetPublisherInfoFunction::Run() {
+  std::unique_ptr<brave_rewards::GetPublisherInfo::Params> params(
+      brave_rewards::GetPublisherInfo::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+      RewardsServiceFactory::GetForProfile(profile);
+
+  if (!rewards_service) {
+    return RespondNow(Error("Rewards service is not initialized"));
+  }
+
+  rewards_service->GetPublisherInfo(
+      params->publisher_key,
+      base::Bind(
+          &BraveRewardsGetPublisherInfoFunction::OnGetPublisherInfo,
+          this));
+  return RespondLater();
+}
+
+void BraveRewardsGetPublisherInfoFunction::OnGetPublisherInfo(
+    const int32_t result,
+    std::unique_ptr<::brave_rewards::PublisherInfo> info) {
+  auto dict = std::make_unique<base::Value>(base::Value::Type::DICTIONARY);
+
+  if (!info) {
+    Respond(
+        TwoArguments(std::make_unique<base::Value>(result), std::move(dict)));
+    return;
+  }
+
+  dict->SetStringKey("publisher_key", info->id);
+  dict->SetStringKey("name", info->name);
+  dict->SetIntKey("percentage", info->percent);
+  dict->SetIntKey("status", info->status);
+  dict->SetIntKey("excluded", info->excluded);
+  dict->SetStringKey("url", info->url);
+  dict->SetStringKey("provider", info->provider);
+  dict->SetStringKey("favicon_url", info->favicon_url);
+
+  Respond(TwoArguments(std::make_unique<base::Value>(result), std::move(dict)));
+}
+
+BraveRewardsGetPublisherPanelInfoFunction::
+    ~BraveRewardsGetPublisherPanelInfoFunction() {}
+
+ExtensionFunction::ResponseAction
+BraveRewardsGetPublisherPanelInfoFunction::Run() {
+  std::unique_ptr<brave_rewards::GetPublisherPanelInfo::Params> params(
+      brave_rewards::GetPublisherPanelInfo::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+      RewardsServiceFactory::GetForProfile(profile);
+
+  if (!rewards_service) {
+    return RespondNow(NoArguments());
+  }
+
+  rewards_service->GetPublisherPanelInfo(
+      params->window_id,
+      params->media_type,
+      params->publisher_key);
+
+  return RespondNow(NoArguments());
+}
+
+BraveRewardsSavePublisherVisitFunction::
+    ~BraveRewardsSavePublisherVisitFunction() {}
+
+ExtensionFunction::ResponseAction
+BraveRewardsSavePublisherVisitFunction::Run() {
+  std::unique_ptr<brave_rewards::SavePublisherVisit::Params>
+      params(brave_rewards::SavePublisherVisit::Params::Create(
+          *args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+      RewardsServiceFactory::GetForProfile(profile);
+
+  if (!rewards_service) {
+    return RespondNow(NoArguments());
+  }
+
+  rewards_service->SavePublisherVisit(
```

Comment:
please rename to `SavePublisherInfo` so that it matches in shared lib. Also let's add callback for it

Reply:
We already have a `SavePublisherInfo` exposed in mojo that just writes to the database (used in the inline tipping flow), while this function is doing more complicated things (see `Publisher::SaveVisitInternal`, for instance, which this ends up calling).

I could rename the existing mojo function to `SavePublisherInfoForTips` or something like that if we definitely want to make this change (doesn't seem like you can have multiple overloads for mojo functions, so something will need to be renamed to support this I think).

Reply:
but isn't this one just saving publisher info? could we just reuse `SavePublisherInfo`? This save doesn't have any duration in it